documentation/ranking.html

```diff
@@ -45,15 +47,20 @@
 
 
 
-<h2 id="rank-profiles">Rank profiles</h2>
+<h2 id="rank-profile">Rank profile</h2>
 <p>
 Ranking expressions are stored in <a href="reference/search-definitions-reference.html#rank-profile">rank profiles</a>.
-Queries select profile using <a href="reference/search-api-reference.html#ranking.profile">ranking.profile</a> -
-this can be used for implementing different use cases, or bucket testing ranking variations.
+If not specified, the <a href="text-ranking.html">default text ranking</a> profile is used.
 </p><p>
 A rank profile can <em>inherit</em> another rank profile.
 </p><p>
-If rank-profile is not specified, the <a href="text-ranking.html">default text ranking</a> is used.
+Queries select profile using <a href="reference/search-api-reference.html#ranking.profile">ranking.profile</a>,
+or in <a href="searcher-development.html">Searcher</a> code:
+<pre>
+query.properties().set(new CompoundName("ranking.profile"), "my_rank-profile");
```

Comment:
There is a direct programatic api for this, query.getRanking().setProfile("my-rank-profile"), must check syntax.

Reply:
yup, query.getRanking().setProfile("my_rank-profile"); works - thanks!